File: app/src/main/java/br/com/ibm/santander/wallacebaldenebre/ui/base/BaseActivity.java
package br.com.ibm.santander.wallacebaldenebre.ui.base;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity implements MvpContract.MvpView, BaseFragment.Callback {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    protected abstract void setUp();

    protected abstract void casts();
}

File: STDAndroidTeste/app/src/main/java/br/com/cesarsicas/stdandroidteste/base/BaseActivity.kt
package br.com.cesarsicas.stdandroidteste.base

import android.support.v7.app.AppCompatActivity
import br.com.cesarsicas.stdandroidteste.AppComponent
import br.com.cesarsicas.stdandroidteste.MainApplication

/**
 * Created by julio on 4/22/18.
 */
open class BaseActivity : AppCompatActivity()  {
    protected val appComponent: AppComponent
        get() {
            val application = application as MainApplication

            return application.appComponent ?: throw IllegalStateException()
        }
}